src/main/java/org/jabref/logic/importer/fetcher/ISIDOREFetcher.java

```diff
@@ -0,0 +1,223 @@
+            }
+        }
+        return "";
+    }
+
+    // Get the type of the document, ISIDORE only seems to have select types, also their types are different to
+    // those used by JabRef.
+    private EntryType getType(NodeList list) {
+        for (int i = 0; i < list.getLength(); i++) {
+            String type = list.item(i).getTextContent();
+            if (type.contains("article") || type.contains("Article")) {
+                return StandardEntryType.Article;
+            }
+            if (type.contains("thesis") || type.contains("Thesis")) {
+                return StandardEntryType.Thesis;
+            }
+            if (type.contains("book") || type.contains("Book")) {
+                return StandardEntryType.Book;
+            }
+        }
+        return StandardEntryType.Misc;
+    }
+
+    // Gets all the authors, separated with the word "and"
+    // For some reason the author field sometimes has extra numbers and letters.
+    private String getAuthor(Node itemElement) {
+        boolean singleAuthor = true;
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 1; i < itemElement.getChildNodes().getLength(); i += 2) {
+            String next = removeNumbers(itemElement.getChildNodes().item(i).getTextContent()).replaceAll("\\s+", " ");
+            next = next.replace("\n", "");
+            if (next.isBlank()) {
+                continue;
+            }
+            if (singleAuthor) {
+                singleAuthor = false;
+            } else {
+                stringBuilder.append(" and ");
+            }
+            stringBuilder.append(next);
+        }
+        return (stringBuilder.substring(0, stringBuilder.length())).trim().replaceAll("\\s+", " ");
+    }
+
+    // Remove numbers from a string and everything after the number, (helps with the author field).
+    private String removeNumbers(String string) {
+        for (int i = 0; i < string.length(); i++) {
+            if (Character.isDigit(string.charAt(i))) {
+                return string.substring(0, i);
+            }
+        }
+        return string;
+    }
+
+    // In the XML file the publishers node often lists multiple publisher e.g.
+    // <publisher origin="HAL CCSD">HAL CCSD</publisher>
+    // <publisher origin="Elsevier">Elsevier</publisher>
+    // Therefore this function simply gets all of them.
+    private String getPublishers(Node itemElement) {
+        if (itemElement == null) {
+            return "";
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+        for (int i = 0; i < itemElement.getChildNodes().getLength(); i++) {
+            stringBuilder.append(itemElement.getChildNodes().item(i).getTextContent().trim());
+            if (itemElement.getChildNodes().item(i).getTextContent().isBlank()) {
+                continue;
+            }
+            stringBuilder.append(", ");
+        }
+        return stringBuilder.substring(0, stringBuilder.length() - 2);
```

Comment:
```suggestion
        return stringBuilder.substring(0, stringJoiner.toString());
```

You don't need to remove the last two character with StringJoiner. #convenience